jerry-core/jerry.c

```diff
@@ -1805,7 +1816,14 @@ jerry_run_simple (const jerry_api_char_t *script_source, /**< script source */
   {
     if ((flags & JERRY_FLAG_PARSE_ONLY) == 0)
     {
-      ret_code = jerry_run (&error_obj_p);
+      jerry_api_value_t error_value;
+      ret_code = jerry_run (&error_value);
+
+      if (jerry_api_value_is_object (&error_value)
+          || jerry_api_value_is_string (&error_value))
```

Comment:
These checks seem to be superfluous here since `jerry_api_release_value` repeats them (and is a no-op if its argument is not an object or a string). AFAICR that was the point of introducing `jerry_api_release_value`.